kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -54,62 +54,103 @@ class KyuubiSessionImpl(
   }
 
   val engine: EngineRef = new EngineRef(sessionConf, user)
+  val engineSyncInit = sessionConf.get(SESSION_ENGINE_SYNC_INIT)
+  var engineInitOp: Operation = _
```

Comment:
private val engineInitOp = essionManager.operationManager.newLaunchEngineOperation(this)